amlip_cpp/src/cpp/types/model/ModelStatisticsDataType.cpp

```diff
@@ -47,22 +47,65 @@ const char* ModelStatisticsDataType::TYPE_NAME_ = "AMLIP-MODEL-STATISTICS";
 
 ModelStatisticsDataType::ModelStatisticsDataType()
     : name_("ModelStatisticsDataTypeName")
+    , data_(nullptr)
+    , data_size_(0)
     , has_been_allocated_(false)
 {
+    // Do nothing
 }
 
 ModelStatisticsDataType::ModelStatisticsDataType(
         const char* name)
     : name_(name)
+    , data_(nullptr)
+    , data_size_(0)
     , has_been_allocated_(false)
 {
+    // Do nothing
 }
 
 ModelStatisticsDataType::ModelStatisticsDataType(
         const std::string& name)
     : name_(name)
+    , data_(nullptr)
+    , data_size_(0)
     , has_been_allocated_(false)
 {
+    // Do nothing
+}
+
+ModelStatisticsDataType::ModelStatisticsDataType(
+        const std::string& name,
+        void* data,
+        const uint32_t size)
+    : name_(name)
+    , data_(data)
+    , data_size_(size)
+    , has_been_allocated_(true)
```

Comment:
I don't think this is correct; ``has_been_allocated`` means that this object has allocated resources, and should hence free them on destruction.
As discussed offline, I believe we should add an extra boolean argument `copy_data` to determine whether to copy the content or just the pointer. This base constructor should be called in the overloaded ones (bytes and string).